sdks/typescript/src/apache_beam/io/kafka.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as beam from "../../apache_beam";
+import * as external from "../transforms/external";
+import { Schema } from "../proto/schema";
+import { RowCoder } from "../coders/row_coder";
+import { serviceProviderFromJavaGradleTarget } from "../utils/service";
+import * as protobufjs from "protobufjs";
+import { camelToSnakeOptions } from "../utils/utils";
+
+const KAFKA_EXPANSION_GRADLE_TARGET =
+  "sdks:java:io:expansion-service:shadowJar";
+
+export type ReadFromKafkaOptions = {
+  keyDeserializer?: string;
+  valueDeserializer?: string;
+  startReadTime?: number;
+  maxNumRecords?: number;
+  maxReadTime?: number;
+  commitOffsetInFinalize?: boolean;
+  timestampPolicy?: "ProcessingTime" | "CreateTime" | "LogAppendTime";
+};
+
+const defaultReadFromKafkaOptions = {
+  keyDeserializer:
+    "org.apache.kafka.common.serialization.ByteArrayDeserializer",
+  valueDeserializer:
+    "org.apache.kafka.common.serialization.ByteArrayDeserializer",
+  timestampPolicy: "ProcessingTime",
+};
+
+export function readFromKafka(
+  consumerConfig: { [key: string]: string }, // TODO: Or a map?
+  topics: string[],
+  options: ReadFromKafkaOptions = {}
+): beam.AsyncPTransform<beam.Root, beam.PCollection<any>> {
```

Comment:
the return type - is it `any`? Can we do something like parameterize it as `T`? I don't know what's best practice in typescript

@KonradJanica - we're trying to have a `Row` type - basically, an element with a schema. I guess in JSON these are just objects - is there a better way of delimiting them rather than `any`? In this case, it's just a simple JSON-like object without functions, only data attached to it.